### PR TITLE
fix(core): harden crypto input validation and unify secret zeroization

### LINKED ISSUE #715: core: unify secret-zeroization across seal encode/decode/grant paths (panic-safe whole-tree)
Core secret-zeroization is inconsistent across the seal encode/decode/grant paths. Surfaced independently by the retro reviews of #680, #673, and #675 (two reviewers flagged the grant.rs gap).

#680 landed a correct, panic-safe `ScrubOnDrop` + whole-tree `Value::zeroize_bytes()` for `encode_read_body`. The same rigor is not applied elsewhere:

1. **`decode_read_body` leaves an un-zeroized transient `Value` tree** (from #673) — `crates/core/src/seal/body.rs:291-327`. `decode(bytes)` allocates a `Value` tree holding a verbatim copy of every file version's 32-byte content key; the bytes are read into zeroizing `SecretBytes`, but the intermediate tree is dropped WITHOUT wiping. Asymmetric with the carefully-scrubbed encode side. **Fix:** wrap the transient `value` in `ScrubOnDrop` (or call `value.zeroize_bytes()` after building `ReadBody`), covering Ok and Err.

2. **`grant.rs zeroize_secret_bytes` is fragile and not panic-safe** (from #680 AND #675) — `crates/core/src/seal/grant.rs:57-65`, call sites `231,361,591`. It wipes only explicitly-named keys and runs as a plain trailing statement after `encode` (a panic in `encode` skips it entirely; a new secret field added without appending its key silently regresses). **Fix:** replace with the whole-tree pattern — `let mut value = Value::Map(m); let out = encode(&value); value.zeroize_bytes(); out` — retiring the bespoke helper + per-key list. (Also a `/simplify` win.)

3. **(Low) filename `Value::Text` copies not scrubbed** (from #680) — `value.rs:141-152` wipes `Value::Bytes` only; transient `Value::Text` copies of filenames (user-private metadata in a ZK system) are dropped un-scrubbed. Consider extending `zeroize_bytes` to `Text` for whole-tree metadata hygiene.

Note: the separate encode-buffer *reallocation* leak is already addressed by open PR #689 / issue #681.

### PR DESCRIPTION
**Pull request by FSM1**
Four batched, additive crypto-hardening changes in `crates/core`, from the retro reviews of #664/#673/#675/#680. Every change only appends new reject vectors — no existing accept vector's bytes change, and there is no wire change.

## 1. Panic-safe whole-tree secret zeroization (#715)
`decode_read_body` now wraps its transient decoded `Value` tree (a verbatim copy of every version's content key) in an owning `ScrubOwned` guard that wipes on Ok, Err, and panic-unwind. The grant/owner/history encode helpers drop the bespoke per-key `zeroize_secret_bytes` allowlist for the shared `ScrubOnDrop` whole-tree guard (a panic in `encode` can no longer skip the wipe; a new secret field can no longer silently regress). `Value::zeroize_bytes` now also scrubs `Value::Text` (transient filename copies). The guard only ever touches the codec's own transient copies — caller `SecretBytes` inputs are untouched.

## 2. Reject non-33-byte SEC1 identity keys + document the aead ceiling (#709)
`EcdsaVerifier::from_sec1` guards the frozen 33-byte compressed width before `from_sec1_bytes` (which also accepts 65-byte uncompressed/hybrid). A compressed key re-emitted uncompressed decodes to the same point but is byte-distinct — a `PartialEq`-equal contact code with different bytes, and in the mailbox path (where `sig_preimage` hashes the raw sender key) a distinct signed preimage that still verified. Also documents the XChaCha20-Poly1305 plaintext ceiling as a caller invariant (`MAX_PLAINTEXT_LEN`) instead of cascading the unreachable ~256 GiB case into six seal signatures.

## 3. Reject low-order X25519 points and non-contributory ECDH (#708)
`X25519Public::from_bytes` is now fail-closed (rejects the RFC 7748 small-order encodings), so holding an `X25519Public` means "not low-order" and `dhkem_encap` stays infallible. `diffie_hellman` returns `None` on a non-contributory result (the exhaustive backstop). `dhkem_decap` returns `Result` and `hpke_open` propagates the new trust check `hpke-non-contributory`; contact import routes a low-order `encSubkey` to the same check (a chosen-key attack, not a parse error). RFC 9180 §7.1.4; the A.1 conformance vectors still pass byte-for-byte.

## 4. Reject duplicate blinded tags in the grant ledger + commitment (#710)
Neither `decode_write_body` nor `decode_grant_set_commitment` rejected a duplicate recipient blinded `tag` (the CBOR duplicate-key check can't see across separate maps in an array). A shared-write holder could inject a second ledger row with a victim's tag, `Permission::Write`, and an attacker-chosen `recipientEncPk` — a confused-deputy over read-vs-write authority. New trust check `duplicate-grant-tag` (the analog of `duplicate-id`), enforced by a `BTreeSet` pass at both decode sites and mirrored by an encode-side `debug_assert`.

## New reject KAT vector families/names
- `contact/reject.json`: `identity-pk-uncompressed` → `invalid-identity-key`; `enc-subkey-low-order` → `hpke-non-contributory`
- `payload/mailbox_reject.json`: `sender-pk-uncompressed` → `identity-signature-invalid`
- `hpke/open_reject.json`: `low-order-enc` → `hpke-non-contributory`
- `grant/write_body_reject.json`: `ledger-duplicate-tag` → `duplicate-grant-tag`
- `grant/grant_set_reject.json`: `entries-duplicate-tag` → `duplicate-grant-tag`

Plus unit tests: uncompressed-key rejection, RFC 7748 low-order table, aead ceiling pin, and duplicate-tag rejection in both grant families.

## Testing
All core targets green: `cargo test -p cipherbox-core` (142 lib + 56 kat_manifest + 6 codec_props + 5 seal_props + 2 kdf_props + 2 ipns_props), `cargo clippy -p cipherbox-core --all-targets` clean, `kat_gen` regenerated and committed. Self-reviewed against simplification, security, and crypto-privacy lenses.

Closes #715, #709, #708, #710

## Summary by CodeRabbit

* **Security Improvements**
  * Rejects low-order X25519 keys and non-contributory HPKE inputs.
  * Rejects non-canonical identity key encodings.
  * Detects duplicate grant tags during encoding and decoding.
  * Strengthens sensitive data cleanup for decoded content.

* **Bug Fixes**
  * Correctly reports invalid identity signatures and trust violations.
  * Adds safeguards for oversized encrypted payloads.

* **Tests**
  * Expanded cryptographic edge-case coverage and validation vectors.

### CODE REVIEW

File: crates/core/src/seal/grant.rs [hunk 1]
```diff
@@ -720,9 +709,30 @@ pub fn decode_grant_set_commitment(bytes: &[u8]) -> Result<GrantSetCommitment, C
...
 /// preimage the owner ECDSA-signs and a recipient verifies.
+///
+/// Encoding does not re-check tag uniqueness; a `debug_assert` catches a
+/// caller-built commitment with duplicate tags early, while its bytes still
+/// reject on decode.
 pub fn encode_grant_set_commitment(c: &GrantSetCommitment) -> Vec<u8> {
+    debug_assert!(
+        assert_entry_tags_unique(&c.entries).is_ok(),
+        "encoding a grant-set commitment with duplicate tags; its bytes would reject on decode"
+    );
```
[the remark this answers is not in the document]
Comment (FSM1) [reply crates/core/src/seal/grant.rs]: Fixed in 43f72cdc6. `encode_grant_set_commitment` is now fallible and runs the same `duplicate-grant-tag` reject release-active (not a `debug_assert`); `sign_grant_set` propagates it, so a release build can no longer sign+publish a commitment its own decoder rejects. Added `encode_and_sign_reject_duplicate_tags` covering both. No wire change, KATs unchanged.

File: crates/core/src/seal/write_body.rs [hunk 2]
```diff
@@ -207,9 +210,32 @@ pub fn decode_write_body(bytes: &[u8]) -> Result<WriteBody, CodecError> {
...
 }
 
+/// Fail-closed uniqueness over a grant ledger's blinded tags — the exact analog
+/// of the read-body child-id uniqueness (#39 D7): a recipient's tag names at most
+/// one grant, so a duplicate is a confused-deputy over read-vs-write authority.
+fn assert_ledger_tags_unique(entries: &[GrantLedgerEntry]) -> Result<(), CodecError> {
+    let mut tags = BTreeSet::new();
+    for e in entries {
+        if !tags.insert(e.tag) {
+            return Err(TrustViolation::DuplicateGrantTag.into());
```
[the remark this answers is not in the document]
Comment (FSM1) [reply crates/core/src/seal/write_body.rs]: Fixed in 43f72cdc6. `encode_write_body` is now fallible and runs the same `duplicate-grant-tag` reject release-active (not a `debug_assert`). Unlike read-body (validated at the `seal_read_body` core boundary), the write-body is sealed outside core, so the encoder is its own release-active guard and never emits a ledger its decoder rejects. Added `encode_rejects_duplicate_ledger_tags`. No wire change, KATs unchanged.

File: crates/core/src/seal/grant.rs [hunk 3]
```diff
@@ -43,27 +43,15 @@ use crate::suite::secret::{SECRET_LEN, SecretBytes};
...
 use super::aad::{AadContext, build_aad};
-use super::body::{bytes_fixed, collect_unknown, merge_unknown, req};
+use super::body::{
+    ScrubOnDrop, assert_grant_tags_unique, bytes_fixed, collect_unknown, merge_unknown, req,
+};
 
 /// The HPKE `info` for every grant-section seal. The structured AAD already
 /// binds `(v, id, scope, epoch, structTag)`, so no additional info label is
 /// needed; it is fixed empty and frozen by the KAT.
 const GRANT_HPKE_INFO: &[u8] = b"";
```
[the remark this answers is not in the document]
Comment (FSM1) [reply crates/core/src/seal/grant.rs:55]: Done in 1ca0a33ee. All three decoders (decode_grant_blob_payload, decode_override_seed_payload, decode_history_link_payload) carry secret material in their transient tree (scope seeds / pointer read key / override seed / prev-epoch seed) and now wrap it in ScrubOwned, matching decode_read_body. The guard wipes only the decoder-owned transient; the produced SecretBytes outputs are untouched (terminal-owner rule). Added decode_grant_blob_payload_scrub_preserves_secret_outputs to lock the Ok path. No KAT diff (memory-hygiene only).

**Comment by FSM1** [comment]
Both review-body nitpicks folded in 1ca0a33ee (they were in the review body, not resolvable threads):

- **write_body.rs `duplicate_ledger_tag_rejects`** — reworded to drop the stale "bypasses the encode-side debug_assert" claim (encode now runs the release-active `assert_grant_tags_unique`), keeping the real rationale that the bytes are hand-built wire data that never passed through any encoder, so decode is what must reject them.
- **kat_gen.rs low-order constant** — hoisted a single module-level `LOW_ORDER_X25519` above `build_hpke_open_reject` and reused it in both call sites; removed the duplicate definition. Confirmed the two byte arrays were byte-identical before merging. No KAT diff.
